### PR TITLE
added gender ratio

### PR DESCRIPTION
**Pull request by fabe85**
@jonas-he 
@swathi-ssunder 
@samitsv 
@vivek-sethia

### CODE REVIEW

File: app/models/pokemon.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ var pokemonSchema = new Schema({
     userId     : {type: String},
     deviceId   : {type: String},
     category   : {type: [Boolean]},
-    gender     : {type: String, enum: ['m', 'f', 'u', 'h']},
+    gender     : {abbreviation: {type: String, enum: ['m', 'f', 'u', 'h']}, male_ratio: Number, female_ratio: Number},
```
Comment (sacdallago) [inline app/models/pokemon.js]: :D Happy to see that you correct the code even before I come in with comments :D

**Comment by sacdallago** [comment]
I don't understand the logic of the ratio? Are there pokemons that are more male than female? Is it an interval instead of a nominal value?
@PokemonGoers/pokedata

**Comment by fabe85** [comment]
@sacdallago According to http://bulbapedia.bulbagarden.net/wiki/List_of_Pok%C3%A9mon_by_gender_ratio there are in fact sometimes more female pokemons than male ones and vice versa. It is, however, always a fixed ratio, i.e. for instance 1:7, so that for 1 male pokemon there are 7 female ones.

**Comment by sacdallago** [comment]
HA! cool! This is some solid good information! Well done.
I would though also expand the data model to have unbreedable pokemon?

Can you help me interpret some results?
- https://github.com/PokemonGoers/PokeData/pull/47/files#diff-946252d0f727c56ab4ae7f0dff545879R1197 Undefined sex but still 1 male and 1 female component thus 100% breedable?
- https://github.com/PokemonGoers/PokeData/pull/47/files#diff-946252d0f727c56ab4ae7f0dff545879R1189 Hermaphrodite with 0 to 0 count, thus can take any sex but cannot reproduce?

In my view you can engineer this feature a bit differently, better. For instance:

``` javascript
...
sex: someSex,
....
```

If `someSex` is `h`, then nothing added (because for every 2 pokemon, they always have a 100% chance of breeding **AT LEAST IN CONVENTIONAL LOGIC**, if this doesn't apply than use the scheme below), else

``` javascript
...
breedable: {
  female: [0, ∞]
  male: [0, ∞]
},
....
```

where female is the ratio between females and males, male is male/female and any instance of `[0, ∞]` `[0, 0]`, `[∞, 0]` represents unbreedable pokemons.
Rationale: Values > 1 have high probability of breeding

**Comment by fabe85** [comment]
I would rather suggest to add the possibility of being breedable as a boolean.

There are some  pokemons who are genderless but are not unbreedable (I don't know why), e.g. Magnemite is supposed to be genderless (so 0:0 ratio) but it is not listed as an unbreedable pokemon.
On the other side, there are pokemons which are only female and can reproduce and others which cannot. But, what is very strange, all pokemons which are only male can reproduce.
So, I think it is not logic in conventional way.

However, the ratio of male and female proportion of a certain pokemon is always fix. But I don't think
that we conclude anything about the fertility or distribution.

Furthermore, when saying a pokemon's gender is undefined that means that we cannot say if it is male or female. I used h (standing for hermaphrodite) for pokemons which are genderless because there are no hermaphrodite pokemons.

**Comment by sacdallago** [comment]
@fabe85 

> Furthermore, when saying a pokemon's gender is undefined that means that we cannot say if it is male or female. I used h (standing for hermaphrodite) for pokemons which are genderless because there are no hermaphrodite pokemons.

This sounds like it should be exactly the opposite :joy: hermaphrodite is: it has a sex, but no idea which one. undefined is undefined; no sex!

> There are some pokemons who are genderless but are not unbreedable (I don't know why), e.g. Magnemite is supposed to be genderless (so 0:0 ratio) but it is not listed as an unbreedable pokemon.
> On the other side, there are pokemons which are only female and can reproduce and others which cannot. But, what is very strange, all pokemons which are only male can reproduce.
> So, I think it is not logic in conventional way.

I feared that this was the case. No conventional logic then. But as you say, does it then make sense to store this ratio? Is it informative in any way? If it has no consequences on anything else, it's just purely statistical and finalized to itself, right?
I'm just asking because from the perspective of data collection: yes, this should stay. But when we will use this data for @PokemonGoers/predictpokemon-1 @PokemonGoers/predictpokemon-2 this might turn out to be a completely random/uninformative/useless feature ( @goldbergtatyana ).

On the other hand "breedability" seems to be an interesting feature! Because breedable pokemons seem to appear more rarely than others.

**Comment by jonas-he** [comment]
So i played Pokemon GO since the release and there is **NO** such thing as gender or gender ratio currently in the game. However if our info page should not just display information regarding Pokemon GO but regarding the whole Pokemon Universe then you can keep this feature.

Note: gender and gender ratio are a thing in the Gameboy games.

Another thing: in Pokemon GO you can breed eggs (by walking a certain distance) which will then turn into pokemon and the 2km eggs can give you crappy pokemon which you find on every corner, the 5km eggs give you a bit more rare pokemon and the 10km ones give the very rare pokemon, see http://imgur.com/gallery/pPDl1#9HdbtXA
To close up my point: there is also no gender attached to the breeded pokemon

**Comment by gyachdav** [comment]
@jonas-he: as @sacdallago mentioned, this info may only be important as features for the machine learning part. The info page should be restricted to data that is _only_ relevant to the Pokemon Go page. So gender and gender ratio is irrelevant. 

Just FYI: I also saw somewhere someone saying that there is a 1:7 female:male ratio. However look here: http://bulbapedia.bulbagarden.net/wiki/Clefairy_(Pok%C3%A9mon) the ration here is 1:3 m:f.

**Comment by fabe85** [comment]
There are basically 7 cateogories of gender ratio (only male/female, genderless, 1:7, 1:3, 1:1, 3:1.

I don't know if there is a connection between gender and appearance or whatsoever, but I think that pokemons like unbreedable genderless pokemons (mew or mewto) are extremely rare. Maybe it is not 
so obvious in the first sight but I can imagine that for ML it could play a role.

@goldbergtatyana also asked in a comment of the former branch to include a gender feature.

I am open for every suggestion, but I think that the more data the better for ML and in terms of 
appearance this gender feature could make a difference in the end.

**Comment by jonas-he** [comment]
@gyachdav wait ... so you're telling me that gender may be relevant for ML to predict spawns when there is no gender in Pokemon Go? I have some doubts on that ... but I'm not angry if someone proves me wrong :smile:  I would have said that it's more relevant for the info page (if we want to cover the whole Pokemon Universe).

**Comment by gyachdav** [comment]
@jonas-he well the fact that gender does not appear in the game does not mean the algorithm responsible for spawning poekmons is not taking gender into account. that is why we're trying this out. 

And no, we are not covering the whole pokemon universe in the info page. Just the Pokemon Go game.

**Comment by jonas-he** [comment]
Ok so gender not relevant for info page, but could be relevant for ML.
Got it.

**Comment by sacdallago** [comment]
@jonas-he @fabe85 

> I don't know if there is a connection between gender and appearance or whatsoever, but I think that pokemons like unbreedable genderless pokemons (mew or mewto) are extremely rare. Maybe it is not 
> so obvious in the first sight but I can imagine that for ML it could play a role.

The rationale that unbreedability is useful for the predictions is totally correct. I would **not** get rid of this feature, or the ratios or the sex.

@gyachdav 

> The info page should be restricted to data that is only relevant to the Pokemon Go page. So gender and gender ratio is irrelevant.

I would integrate Pokemon Go with additional sources to give a more descriptive info page, more on the lines of what we did last semester. Anyway this decision we can make later with the @PokemonGoers/catch-em-all group. But for now please collect everything. ( We mentors will talk about this offline, update will follow)

**Comment by jonas-he** [comment]
Note that the breeding values @fabe85 used are from the Pokemon Gameboy games.
For Pokemon GO refer to https://www.primagames.com/media/files/Pokemon-Go-eggs.jpg/PRIMAP/resize/1200x/quality/80

**Pull request merged**